app/models/concerns/gbv_key_performance_indicators.rb

```diff
@@ -0,0 +1,257 @@
+# frozen_string_literal: true
+
+# Module for capturing all of the GBV KPI related logic.
+module GBVKeyPerformanceIndicators
```

Comment:
Not happy with this architecture either. It feels like a Form / Response is an underlying concept that wants to be drawn out and captured in a query-able object tree but I'm not certain how yet.

Reply:
I don't mind this so much: You have some GBV specific business logic that you cordoned off into a concern. Another option is to create a service object:

```
class GbvKpiCalculations

  def psychosocial_goals_progress
    ...
  end

  def justice_goals_progress
   ...
  end
  ....
end